### PR TITLE
feat: cleanup errors

### PR DESCRIPTION
**Pull request by SWvheerden**
Description
---
This ensures that we do not have any unused error types left in the validation errors. 
Renames some old errors to more clearly represent the actual error. 

Motivation and Context
---
Some errors where unused while some where sending the wrong error type back

### CODE REVIEW

File: base_layer/core/src/base_node/sync/block_sync/synchronizer.rs [hunk 1]
```diff
@@ -306,15 +305,12 @@ impl<B: BlockchainBackend + 'static> BlockSynchronizer<B> {
...
-            .map_err(|err| ValidationError::CustomError(err.to_string()))?;
+            .await?;
 
             let block = match res {
                 Ok(block) => block,
                 Err(err @ ValidationError::BadBlockFound { .. }) |
                 Err(err @ ValidationError::FatalStorageError(_)) |
-                Err(err @ ValidationError::AsyncTaskFailed(_)) |
-                Err(err @ ValidationError::CustomError(_)) => return Err(err.into()),
                 Err(err) => {
```
Comment (stringhandler) [inline base_layer/core/src/base_node/sync/block_sync/synchronizer.rs:315]: This changes the logic, are you sure it's correct?

File: base_layer/core/src/base_node/sync/header_sync/validator.rs [hunk 2]
```diff
@@ -138,8 +138,7 @@ impl<B: BlockchainBackend + 'static> BlockHeaderSyncValidator<B> {
             // We dont want to mark a block as bad for internal failures
             Err(
                 e @ ValidationError::FatalStorageError(_) |
-                e @ ValidationError::IncorrectNumberOfTimestampsProvided { .. } |
-                e @ ValidationError::AsyncTaskFailed(_),
```
Comment (stringhandler) [inline base_layer/core/src/base_node/sync/header_sync/validator.rs:142]: Another logic change, are you sure it's correct?

**Review by stringhandler — COMMENTED** [review]
Nice work, but I'm worried that AsyncTaskError is now handled in a different match arm. Was that intended?

File: base_layer/core/src/base_node/sync/header_sync/validator.rs (hunk 2 above)
Comment (SWvheerden) [reply base_layer/core/src/base_node/sync/header_sync/validator.rs:142]: This is correct. 
This only suppresses internal errors to let them be the reason a block is flag as a bad block

File: base_layer/core/src/base_node/sync/block_sync/synchronizer.rs (hunk 1 above)
Comment (SWvheerden) [reply base_layer/core/src/base_node/sync/block_sync/synchronizer.rs:315]: This is wrong, yes. BadBlockFound and FatalStorageError should still result in return err.

**Review by stringhandler — APPROVED** [review]

**Pull request merged**